modules/Microsoft.ContainerService/managedClusters/deploy.bicep

```diff
@@ -553,6 +562,40 @@ module managedCluster_agentPools 'agentPools/deploy.bicep' = [for (agentPool, in
   }
 }]
 
+module managedCluster_fluxExtension '../../Microsoft.KubernetesConfiguration/extensions/deploy.bicep' = if (!empty(fluxConfigurations)) {
+  name: '${uniqueString(deployment().name, location)}-ManagedCluster-FluxExtension'
+  params: {
+    clusterName: managedCluster.name
+    name: 'flux'
+    extensionType: 'microsoft.flux'
+    enableDefaultTelemetry: enableReferencedModulesTelemetry
+    location: location
+    releaseNamespace: 'flux-system'
+    releaseTrain: !empty(fluxReleaseTrain) ? fluxReleaseTrain : 'Stable'
+    version: !empty(fluxVersion) ? fluxVersion : ''
+  }
+}
+
+module managedCluster_fluxConfiguration '../../Microsoft.KubernetesConfiguration/fluxConfigurations/deploy.bicep' = [for (fluxConfiguration, index) in fluxConfigurations: {
```

Comment:
Note: Test if it may make sense to either move the `extension` into the `fluxConfiguration` or vice versa

Reply:
@AlexanderSehr Opened a new PR for this approach: https://github.com/Azure/ResourceModules/pull/2679